File: Code/PS2/Python/qdecode.py
def qdecode(code, dtable):
    """
    Quickly decodes a previously encoded number into a weakly descending n-tuple.

    Args:
        code (int): Encoded integer state code.
        dtable (numpy.ndarray): Decoding lookup table.

    Returns:
        numpy.ndarray: Decoded weakly descending n-tuple.
    """
    return dtable[:, code]  # Retrieve the n-tuple from the decoding table

File: Code/PS2/Matlab/qdecode.m
function [out1] = qdecode(code,dtable)
% This procedure does a quick decode of a previously encoded number into
% a weakly descending n-tuple. Decoding is done using a table lookup. Each
% column of the table consists of an n-tuple; the ith column is the ith
% n-tuple to be decoded. The table is stored in the variable "dtable".
% global dtable
out1 = dtable(:,code);
end